### PR TITLE
juniper northstar compatibility added

### PR DESCRIPTION
**Pull request by n1654**
in order to make connection to Juniper Northstar API
1) data structure added:{"grant_type":"password","username":"'${username}'","password":"'${password}'"}
2) auth_header added: 
(probably "colon" should be removed https://tools.ietf.org/html/rfc6750)

### CODE REVIEW

File: adapters/rest_generic/rest_generic_connect.php [hunk 1]
```diff
@@ -95,10 +95,10 @@ public function send($origin, $rest_cmd) {
...
 		}
-		
+
 		$headers = "";
 		$auth = "";
-	
+
 		echo("auth_mode= ".$this->auth_mode."\n");
                 echo("auth_header= ".$this->auth_header."\n");
                 echo("key= ".$this->key."\n");
```
Comment (hiro-ubiqube) [inline adapters/rest_generic/rest_generic_connect.php:104]: @n1654 
just got a feedback with the NEC.
they tried to connect to real Northstar and found it needed Basic Auth on top of Bearer.
so we might want to add something like 
`$auth = " -u " . $this->sd_login_entry . ":" . $this->sd_passwd_entry;`
here.
Do you think it works?

Comment (Korvuss) [reply adapters/rest_generic/rest_generic_connect.php:104]: it already supports basic auth we just need to be able to use both at the same time

Comment (hiro-ubiqube) [reply adapters/rest_generic/rest_generic_connect.php:104]: @Korvuss Is there any way to change auth_mode by MS or by config profiles anyhow?

Comment (Korvuss) [reply adapters/rest_generic/rest_generic_connect.php:104]: @hiro-ubiqube you can set the Auth_mode in the config variables for the device

Comment (hiro-ubiqube) [reply adapters/rest_generic/rest_generic_connect.php:104]: @Korvuss 
You mean if I set config profiles like below, will MS A use token and MS B&C use Basic auth?
`
config A
 - MS A
 - Template A (AUTH_MODE = token)
config B
 - MS B
 - MS C
 - Template B (AUTH_MODE = Basic)
`

**Pull request merged**